allennlp/data/token_indexers/pretrained_transformer_indexer.py

```diff
@@ -31,17 +31,27 @@ class PretrainedTransformerIndexer(TokenIndexer[int]):
         We use a somewhat confusing default value of ``tags`` so that we do not add padding or UNK
         tokens to this namespace, which would break on loading because we wouldn't find our default
         OOV token.
+    mask_padding_with_zero:  ``bool``, optional (default=True)
+        If set to ``True``, the attention mask will be filled by ``1`` for actual values
+        and by ``0`` for padded values. If set to ``False``, inverts it (``1`` for padded values, ``0`` for
+        actual values)
     """
 
     def __init__(
-        self, model_name: str, namespace: str = "tags", token_min_padding_length: int = 0
+        self,
+        model_name: str,
+        namespace: str = "tags",
+        token_min_padding_length: int = 0,
+        mask_padding_with_zero: bool = True,
```

Comment:
Any particular reason for this option?

Reply:
I implemented this pull request by referring to the following code.

https://github.com/huggingface/transformers/blob/master/src/transformers/data/processors/glue.py

When I looked for a case where mask_padding_with_zero would be False, XLNet hit it.

https://github.com/zihangdai/xlnet/blob/master/classifier_utils.py#L113-L115

I can remove this mask_padding_with_zero if I specify Input_mask. I fixed it that way because it seemed simpler.
https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_xlnet.py#L581-L587